src/NServiceBus.Core/StartableEndpoint.cs

```diff
@@ -61,15 +60,29 @@ public StartableEndpoint(SettingsHolder settings, IBuilder builder, FeatureActiv
             return featureRunner;
         }
 
-        static async Task StartReceivers(List<TransportReceiver> receivers)
+        static async Task InitializeReceivers(List<TransportReceiver> receivers)
+        {
+            foreach (var receiver in receivers)
+            {
+                try
+                {
+                    await receiver.Init().ConfigureAwait(false);
+                }
+                catch (Exception ex)
+                {
+                    Logger.Fatal($"Receiver {receiver.Id} failed to initialize.", ex);
+                    throw;
+                }
+            }
+        }
+
+        static void StartReceivers(List<TransportReceiver> receivers)
         {
             foreach (var receiver in receivers)
             {
-                Logger.DebugFormat("Starting {0} receiver", receiver.Id);
                 try
                 {
-                    await receiver.Start().ConfigureAwait(false);
-                    Logger.DebugFormat("Started {0} receiver", receiver.Id);
```

Comment:
tbh I wasn't sure about the value of these log statements. The one in the transport receiver also mentions the related input queue so you know where to start looking when something happens. But just "starting x" and "started x" doesn't seem to provide any valuable information imho.